moveit_core/robot_state/test/robot_state_benchmark.cpp

```diff
@@ -32,132 +32,338 @@
  *  POSSIBILITY OF SUCH DAMAGE.
  *********************************************************************/
 
-/* Author: Robert Haschke */
+/* Author: Mario Prats, Robert Haschke */
+
+// To run this benchmark, 'cd' to the build/moveit_core/robot_state directory and directly run the binary.
+
+#include <benchmark/benchmark.h>
+#include <kdl_parser/kdl_parser.hpp>
+#include <kdl/chainjnttojacsolver.hpp>
 #include <moveit/robot_model/robot_model.h>
 #include <moveit/robot_state/robot_state.h>
 #include <moveit/utils/robot_model_test_utils.h>
-#include <eigen_stl_containers/eigen_stl_containers.h>
-#include <chrono>
-#include <gtest/gtest.h>
+#include <random_numbers/random_numbers.h>
 
-// Helper class to measure time within a scoped block and output the result
-class ScopedTimer
-{
-  const char* const msg_;
-  double* const gold_standard_;
-  const std::chrono::time_point<std::chrono::steady_clock> start_;
+// Robot and planning group for benchmarks, except for RobotStateUpdate and RobotStateForwardKinematics, which will use
+// the pr2.
```

Comment:
Ultra nitpick -- also, should these be called `DEFAULT_TEST_ROBOT` etc. instead?

```suggestion
// the PR2.
```

Reply:
I changed these to:
```
constexpr char PANDA_TEST_ROBOT[] = "panda";
constexpr char PANDA_TEST_GROUP[] = "panda_arm";
constexpr char PR2_TEST_ROBOT[] = "pr2";
constexpr char PR2_TIP_LINK[] = "r_wrist_roll_link";
```